Source/bmalloc/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.h

```diff
@@ -44,22 +44,32 @@
  * allocated by the user, is limited to 1MB. These overall limits should ensure that the memory impact on the system
  * is minimal, while helping to tackle the problems of catching use after frees and out of bounds accesses.
  *
- * PGM will maintain the metadata of recently deallocated objects. This is beneficial for crash analystics to better 
- * identify the type and reason for a crash. After enough deallocations an object will no longer be considered 
+ * PGM will maintain the metadata of recently deallocated objects. This is beneficial for crash analystics to better
+ * identify the type and reason for a crash. After enough deallocations an object will no longer be considered
  * recently deleted and will have its metadata destroyed.
  */
 
 #ifndef PAS_PROBABILISTIC_GUARD_MALLOC_ALLOCATOR
 #define PAS_PROBABILISTIC_GUARD_MALLOC_ALLOCATOR
 
-#include "pas_utils.h"
-#include "pas_large_heap.h"
-#include "pas_ptr_hash_map.h"
 #include <stdbool.h>
 #include <stdint.h>
 
+#include "pas_large_heap.h"
+#include "pas_large_map_entry.h"
+#include "pas_ptr_hash_map.h"
+#include "pas_utils.h"
+
 PAS_BEGIN_EXTERN_C;
 
+#define PGM_BACKTRACE_MAX_FRAMES 31
+
+/* structure for holding the allocation and deallocation backtraces */
+typedef struct {
+    int frame_size;
+    void* backtrace_buffer[PGM_BACKTRACE_MAX_FRAMES];
+} backtrace_metadata_t;
```

Comment:
libpas does not use this form. Define struct backtrace_metadata and use it as " struct backtrace_metadata" Also it should be prefixed with pas_